datasource.go

```diff
@@ -82,6 +85,27 @@ func (ds *sqldatasource) QueryData(ctx context.Context, req *backend.QueryDataRe
 
 }
 
+func (ds *sqldatasource) getDB(q *Query) (*sql.DB, string, error) {
+	// The database connection may vary depending on query arguments
+	// The raw arguments are used as key to store the db connection in memory so they can be reused
+	key := ""
+	db := ds.dbConnections[key]
+	var err error
+	if len(q.Args) != 0 {
+		key = string(q.Args[:])
+		if cachedDB, ok := ds.dbConnections[key]; ok {
+			db = cachedDB
+		} else {
+			db, err = ds.c.Connect(ds.settings, q.Args)
+			if err != nil {
+				return nil, "", err
+			}
+			ds.dbConnections[key] = db
+		}
+	}
+	return db, key, nil
+}
```

Comment:
```suggestion
	// The database connection may vary depending on query arguments
	// The raw arguments are used as key to store the db connection in memory so they can be reused
	key := DefaultKey
	db := ds.dbConnections[key]
	if len(q.Args) == 0 {
		return db, key, nil
	}
	
	key = string(q.Args[:])
	if cachedDB, ok := ds.dbConnections[key]; ok {
		return cachedDB, key, nil
	}
	
	db, err := ds.c.Connect(ds.settings, q.Args)
	if err != nil {
		return nil, "", err
	}
	
	// Assign this connection in the cache
	ds.dbConnections[key] = db
	
	return db, key, nil
}
```

Always prefer early returns over stacking else blocks.